ranger/default.go

```diff
@@ -97,331 +95,151 @@ var (
+		resp.WithRootUrl(url.String()),
+		resp.WithUnauthTemplate(defaultUnauthedTmpl),
+		resp.WithVueTemplate(defaultVueTmpl),
 	}
-}
 
-// DefaultRouter constructs a RangerOption that returns a followup option
-// configuring the *Ranger.Router to be used by the web server.
-func DefaultRouter() RangerOption {
-	return func(rng *Ranger) (OptFollowup, error) {
-		return func() error {
-			fn, err := DefaultResponder()(rng)
-			if err != nil {
-				return err
-			}
-
-			if err := fn(); err != nil {
-				return err
-			}
-
-			mws := make([]middleware.Adapter, 0)
-			if rng.env.IsProduction() {
-				mws = append(
-					mws,
-					middleware.ForceHTTPS(rng.env.String()),
-				)
-			}
-
-			mws = append(
-				mws,
-				middleware.RequestID(defaultRequestIDCtxKey),
-				middleware.InjectIPAddress(),
-				middleware.LogRequest(rng.Logger),
-			)
-
-			if rng.sessions == nil {
-				if _, err = DefaultSessionStore()(rng); err != nil {
-					return err
-				}
-			}
-
-			mws = append(
-				mws,
-				middleware.InjectSession(rng.sessions, rng.kr.SessionKey()),
-				middleware.CurrentUser(
-					rng.Responder,
-					rng.userstore,
-					rng.kr.SessionKey(),
-					rng.kr.CurrentUserKey(),
-				),
-			)
-
-			r := router.NewRouter(rng.env.String())
-			r.OnEveryRequest(mws...)
-			r.HandleNotFound(http.HandlerFunc(func(wx http.ResponseWriter, rx *http.Request) {
-				if strings.Contains(rx.Header.Get("Accept"), "text/html") && rx.URL.Path != rng.url.Path {
-					rng.Redirect(wx, rx, resp.ToRoot())
-					return
-				}
-
-				wx.WriteHeader(http.StatusNotFound)
-			}))
-
-			fn, err = WithRouter(r)(rng)
-			if err != nil {
-				return err
-			}
-
-			return fn()
-		}, nil
-	}
+	return resp.NewResponder(args...)
 }
 
-// DefaultSessionStore constructs a RangerOption that configures the SessionStore
-// to be used for storing session data.
-//
-// DefaultSessionStore relies on two env vars:
-// - "SESSION_AUTH_KEY"
-// - "SESSION_ENCRYPTION_KEY"
-//
-// These must be valid hex encoded values.
-//
-// If these values do not exist, DefaultSessionStore will generate new, random keys
-// and save those to the env var file for later use.
-func DefaultSessionStore(opts ...session.ServiceOpt) RangerOption {
-	return func(rng *Ranger) (OptFollowup, error) {
-		args := []session.ServiceOpt{
-			session.WithCookie(),
-			session.WithMaxAge(3600 * 24 * 7),
-		}
-
-		args = append(args, opts...)
-
-		ll := setupLog
-		if sl, ok := ll.(logger.SkipLogger); ok {
-			ll = sl.AddSkip(sl.Skip() + 2)
+// defaultRouter constructs a [router.Router] to be used by the web server.
+func defaultRouter(
+	env trails.Environment,
+	baseURL *url.URL,
+	responder *resp.Responder,
+	mws []middleware.Adapter,
+) router.Router {
+	route := router.NewRouter(env.String())
+	route.OnEveryRequest(mws...)
+	route.HandleNotFound(http.HandlerFunc(func(wx http.ResponseWriter, rx *http.Request) {
+		if strings.Contains(rx.Header.Get("Accept"), "text/html") && rx.URL.Path != baseURL.Path {
```

Comment:
Curious why a redirect was chosen as the default behavior for a `NotFound` (when accepting html) vs the typical `404` response?